File: output/java/Hard/1816-count-of-smaller-numbers-after-self.java
/**https://leetcode.com/problems/count-of-smaller-numbers-after-self/ */
//Given an integer array nums, return an integer array counts where counts[i] is the number of smaller elements to the right of nums[i].


class CountOfSmallerNumbersAfterSelf {
    public List<Integer> countSmaller(int[] nums) {
        int n = nums.length;
        int[] indexes = new int[n];
        for (int i = 0; i < n; i++) {
            indexes[i] = i;
        }
        int[] counts = new int[n];
        mergeSort(nums, indexes, counts, 0, n - 1);
        List<Integer> ans = new ArrayList<Integer>();
        for (int count : counts) {
            ans.add(count);
        }
        return ans;
        
        
    }
}
     
    

File: output/py/Hard/1816-count-of-smaller-numbers-after-self.py
#https://leetcode.com/problems/count-of-smaller-numbers-after-self/
#Given an integer array nums, return an integer array counts where counts[i] is the number of smaller elements to the right of nums[i].


class Solution(object):
    def countSmaller(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        if not nums:
            return []
        res = [0] * len(nums)
        sorted_nums = []
        for i in range(len(nums)-1, -1, -1):
            index = self.bisect_left(sorted_nums, nums[i])
            res[i] = index
            sorted_nums.insert(index, nums[i])
        return res
        
        
    